PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -0,0 +1,209 @@
+    private JTextField workerCoresField;
+    private JTextField workerMemoryField;
+    private JTextField workerNumberOfContainersField;
+    private JTextField availableAUField;
+    private JTextField totalAUField;
+    private JTextField calculatedAUField;
+    private JLabel masterMemoryLabel;
+    private JLabel masterCoresLabel;
+    private JLabel clusterNameLabel;
+    private JLabel adlAccountLabel;
+    private JLabel previousSparkEventsLabel;
+    private JLabel availableAULabel;
+    private JLabel calculatedAULabel;
+    private JLabel workerCoresLabel;
+    private JLabel workerMemoryLabel;
+    private JLabel workerNumberOfContainersLabel;
+    private JTextField errorMessageField;
+    private JPanel provisionDialogPanel;
+    private JButton refreshButton;
+
+    public SparkServerlessProvisionDialog(@NotNull SparkServerlessADLAccountNode adlAccountNode) {
+        // TODO: canBeParent
+        super((Project)adlAccountNode.getProject(), true);
+
+        this.ctrlProvider = new SparkServerlessClusterProvisionCtrlProvider(
+                this, new IdeaSchedulers((Project)adlAccountNode.getProject()));
+        this.adlAccountNode = adlAccountNode;
+
+        init();
+        this.setTitle("Provision Spark Cluster");
+        errorMessageField.setBackground(this.provisionDialogPanel.getBackground());
+        errorMessageField.setBorder(BorderFactory.createEmptyBorder());
+        this.setModal(true);
+
+        adlAccountField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                // TODO: check validity of ADL account
+                availableAUField.setText(String.valueOf(ctrlProvider.getAvailableAU(adlAccountField.getText())));
+                totalAUField.setText(String.valueOf(ctrlProvider.getTotalAU(adlAccountField.getText())));
+            }
+        });
+        refreshButton.addActionListener(e -> {
+                availableAUField.setText(String.valueOf(ctrlProvider.getAvailableAU(adlAccountField.getText())));
+        });
+        masterCoresField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                updateCalculatedAUField();
+            }
+        });
+        masterMemoryField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                updateCalculatedAUField();
+            }
+        });
+        workerCoresField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                updateCalculatedAUField();
+            }
+        });
+        workerMemoryField.addFocusListener(new FocusAdapter() {
+            @Override
+            public void focusLost(FocusEvent e) {
+                updateCalculatedAUField();
+            }
+        });
+        workerNumberOfContainersField.addFocusListener(new FocusAdapter() {
```

Comment:
May you can create an Array, such as `Arrays.asList(masterCoresField, masterMemoryField,...).forEach(comp -> comp.addFocusListener(...))`